packages/amplify-graphql-api-construct-tests/src/__tests__/backends/sql-configurable-stack/app.ts

```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/env node
+import 'source-map-support/register';
+import { App, Stack, CfnOutput, Duration, RemovalPolicy } from 'aws-cdk-lib';
+// @ts-ignore
+import {
+  AmplifyGraphqlApi,
+  AmplifyGraphqlDefinition,
+  AuthorizationModes,
+  IAmplifyGraphqlDefinition,
+  SqlModelDataSourceDbConnectionConfig,
+  ModelDataSourceStrategySqlDbType,
+  PartialTranslationBehavior,
+} from '@aws-amplify/graphql-api-construct';
+import { AccountPrincipal, Effect, PolicyDocument, PolicyStatement, Role } from 'aws-cdk-lib/aws-iam';
+import { CfnUserPoolGroup, UserPool, UserPoolClient, UserPoolTriggers } from 'aws-cdk-lib/aws-cognito';
+import { Function, Runtime, Code } from 'aws-cdk-lib/aws-lambda';
+
+// #region Utilities
+
+enum AUTH_MODE {
+  API_KEY = 'API_KEY',
+  AWS_IAM = 'AWS_IAM',
+  AMAZON_COGNITO_USER_POOLS = 'AMAZON_COGNITO_USER_POOLS',
+  OPENID_CONNECT = 'OPENID_CONNECT',
+  AWS_LAMBDA = 'AWS_LAMBDA',
+}
+
+enum SCHEMA {
+  DEFAULT = /* GraphQL */ `
```

Comment:
Why would we represent this as an enum vs a string?

Reply:
Both options are viable here. My thought originally was that there could be multiple existing schemas available in the stack which could be used without providing schema through `stack-config` but more like an option. But that could be an overdesign here, as almost every test's schema is different, and it's impossible to store all those schemas in the CDK file.

Reply:
I think maybe a similar question is why enum versus an object with a string field `default`?

Reply:
Do we have to re-define these Auth modes and set a default for schema? Instead we could just fail if either of them is not defined in the test using it.

Reply:
I think we could keep the Auth mode to sync with the stack config interface that is using the `AUTH_TYPE` from AppSync, while schema default could be discarded to avoid overdesign, since schema is already required when defining the stack config interface.